web/src/components/hooks/useDownloadValues.test.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from "react";
+import { QueryClient, QueryClientProvider } from "react-query";
+import { act, renderHook } from "@testing-library/react-hooks";
+import { getValues, useDownloadValues } from "./useDownloadValues";
+
+describe("useDownloadValues", () => {
+  describe("GET", () => {
+    let queryClient;
+    let wrapper;
+    beforeEach(() => {
+      queryClient = new QueryClient();
+      wrapper = function wrapperFunc({ children }) {
+        return (
+          <QueryClientProvider client={queryClient}>
+            {children}
+          </QueryClientProvider>
+        );
+      };
+    });
+    it("calls _getValues", async () => {
+      const fetchValuesSpy = jest.fn(() => Promise.resolve());
+      const testConfig = {
+        _getValues: fetchValuesSpy,
+        _createObjectURL: jest.fn(() => "test"),
+        _revokeObjectURL: jest.fn(() => "test"),
+        appSlug: "test",
+        fileName: "test",
+      };
+
+      const { result, waitFor } = renderHook(
+        () => useDownloadValues(testConfig),
+        {
+          wrapper,
+        }
+      );
+
+      await act(async () => {
+        await result.current.download();
+      });
+      await waitFor(() => result.current.isSuccess);
+
+      const expectedFetchConfig = {
+        appSlug: testConfig.appSlug,
+      };
+
+      expect(result.current.variables).toEqual(undefined);
+      expect(fetchValuesSpy).toHaveBeenCalledTimes(1);
+      expect(fetchValuesSpy).toHaveBeenCalledWith(expectedFetchConfig);
+    });
+    it.todo(
+      "test the rest of this or figure out how to use react-query with it"
```

Comment:
there are a couple of unit test todos. I need to pivot to some other work so I can come back to these next week.